### PR TITLE
cmd/geth, internal/flags: print envvar config source and bad names

### PR DESCRIPTION
**Pull request by karalabe**
One catch with env vars is that they might silently change Geth's behavior, in both directions:

- A set config var might be unexpected to the user and they might not know where it originates from (since it's not specified on the CLI).
- An old / mistyped env var might be present that is not actually consumed by Geth because a name mismatch, which would lead to a silent ignore.

Both cases would be quite tricky to debug. Further it would be a PITA for us to debug user issues when they don't share their env list, and sharing them may not be best security practice.

This PR tries to solve all 3 issues, by listing all the consumed env vars, as well as all the env vars that "seem" like GETH ones, but that are not consumed in the end.

TL;DR:

Screenshot 2023-09-14 at 17 24 26

### CODE REVIEW

File: cmd/geth/main.go [hunk 1]
```diff
@@ -249,7 +249,11 @@ func init() {
 	app.Before = func(ctx *cli.Context) error {
 		maxprocs.Set() // Automatically set GOMAXPROCS to match Linux container CPU quota.
 		flags.MigrateGlobalFlags(ctx)
-		return debug.Setup(ctx)
+		if err := debug.Setup(ctx); err != nil {
+			return err
+		}
+		flags.CheckEnvVars(ctx, app.Flags, "GETH")
```
Comment (holiman) [inline cmd/geth/main.go:255]:
```suggestion
		flags.CheckEnvVars(ctx, app.Flags, "GETH_")
```

maybe?

Comment (karalabe) [reply cmd/geth/main.go:255]: Unsure. I deliberately left it as GETH only so that if you forget the underscore or use a minus or something it still gets caught. That said, it's mostly a heuristic for now.

File: internal/flags/helpers.go [hunk 2]
```diff
@@ -263,3 +265,37 @@ func AutoEnvVars(flags []cli.Flag, prefix string) {
...
+	for _, keyval := range keyvals {
+		key := strings.Split(keyval, "=")[0]
+		if !strings.HasPrefix(key, prefix) {
+			continue
+		}
+		if flag, ok := known[key]; ok {
+			if ctx.Count(flag) > 0 {
+				log.Info("Config environment variable found", "envvar", key, "shadowedby", "--"+flag)
+			} else {
+				log.Info("Config environment variable found", "envvar", key)
```
Comment (holiman) [inline internal/flags/helpers.go:295]: In some cases. cmd line params are sensitive (password for remote influxdb databases, password for `--unlock`), but this will echo them in logs, right? 

Might not be the best idea -- so far, we've been kind rightfully lax about asking people for their logs, because we know they contain nothing interesting. 

Ah wait, it's only the `key`, not the value? If so it's all good!

Comment (karalabe) [reply internal/flags/helpers.go:295]: Yes, only key for the exact reason.

**Review by holiman — APPROVED** [review]
LGTM
